### ISSUE TITLE
Verified user filter not picking up a verified user

### ISSUE THREAD

**Issue by Googolplexbyte** [RomainLK/ytc-filter#37]
I was watching Pochimaru/POCHI-GOYA.channel 's stream: https://www.youtube.com/watch?v=59FpSRLb_1k

And I noticed the "Author is verified" filter wasn't working for "Ninomae Ina'nis Ch. hololive-EN", even though it was working for other verified users in the chat such as "Airani Iofifteen Channel hololive-ID" (comments at 44:05 minute mark and earlier).

I think the streamer might be having the same issue, as she mentioned she would have missed her as well if not for the people in chat pointing her out.

"Ninomae Ina'nis Ch. hololive-EN" appears in chat at the 50:05 minute mark, and is marked as verified in chat. I managed to get her to show up in the ytcFilter by using the filter - "Author name is: Ninomae Ina'nis Ch. hololive-EN" and it marks her as verified in the ytcFilter box then, so I'm not sure why it missed her originally with just the "Author is verified" filter.

I'm guessing it's an issue with the punctuation in her name or something.

**Comment by RomainLK** [comment]
Very strange, I can't reproduce your problem with my usual filters + "Author is verified". ytcFilter does get Ina's message. I will do another test with the full video to see if it's an effect  of the length of the video. Only thing I can't test is a real time chat sadly.

**Comment by RomainLK** [comment]
I can reproduce it when viewing the whole video, so I can investigate your bug

**Comment by RomainLK** [comment]
Ok this bug will be hard to figure out. Somehow, all information of Ina's message has been replaced by the information of another message, except the id.

If you using Firefox, you can help me confirm the bug. ( I haven't managed to reproduce it in Chrome so far )

- Visit about:debugging#/runtime/this-firefox
- Find the extension, then click "Inspect".  This will open a debugger.
- In the console, paste the following code, then press "enter" to execute it:

```javascript
browser.storage.local.get('cache:vcVideo59FpSRLb_1k').then(c => {
  const parsed = JSON.parse(c['cache:vcVideo59FpSRLb_1k'].data)
  const msg = parsed.messages.find(m => m.id === 'CjkKGkNKdWhnSURCck8wQ0ZSbDJnd29kdVVnRXhBEhtDSVNQdDhpOHJPMENGUXpORmdrZEVpOEZlUTA%3D')
  console.log(msg)
})
```

This will output in the console the message which has the same ID as Ina's at 50:05.

**Comment by Googolplexbyte** [comment]
It's brave browser I'm using. Version 1.17.75 Chromium: 87.0.4280.88 (Official Build) (64-bit) Let me know if that helps at all.

**Comment by RomainLK** [comment]
Ok, I did some research on how to get the info I'm looking for in Chromium based browser.

For Brave:

- Visit brave://extensions/
- Enable developer mode
- Click on the "inspect views background page" link for ytcFilter. This will open the debugger.
- In the console, execute this:

```javascript
chrome.storage.local.get('cache:vcVideo59FpSRLb_1k',c => {
  const parsed = JSON.parse(c['cache:vcVideo59FpSRLb_1k'].data)
  const msg = parsed.messages.find(m => m.id === 'CjkKGkNKdWhnSURCck8wQ0ZSbDJnd29kdVVnRXhBEhtDSVNQdDhpOHJPMENGUXpORmdrZEVpOEZlUTA%3D')
  console.log(msg)
})
```

And you can post the result here.

**Comment by Googolplexbyte** [comment]
Sorry it took a minute to get back to you.

The response in the console was: "" undefined
VM14:4 {id: "CjkKGkNKdWhnSURCck8wQ0ZSbDJnd29kdVVnRXhBEhtDSVNQdDhpOHJPMENGUXpORmdrZEVpOEZlUTA%3D", message: "|дﾟ)はい!！！描きまーす！！", author: "Ninomae Ina'nis Ch. hololive-EN", authorType: "", avatarUrl: "https://yt3.ggpht.com/ytc/AAUvwng37V0l-NwF3bu7QA4XmOP5EZFwk5zJE-78OHP9=s32-c-k-c0xffffffff-no-rj-mo", …} ""

**Comment by RomainLK** [comment]
Ok, different result, and your ytcFilter did filter her message properly, which just gives me more questions than answer. Actually nevermind, reading your first message, you already managed to get her after with another filter so the saved messages in your browser won't actually help me.

For now I will keep the bug on backburner, if I have an idea or can reproduce it easily, or more reports. Thanks for all the informations you provided.

**Comment by RomainLK** [comment]
Fixed in 2.1.0 following ticket #42 . In term of implementation, there might still be a little period where ytcFilter is imprecise because  it still uses the Mutation observer for the first few messages. But once it switches to the fetch interceptor, the data used by ytcfilter are no longer dependant of what  the chat display, avoiding a lot of possible error due to Youtube's optimization.
